src/Controller/ModelCollector.php

```diff
@@ -241,6 +266,47 @@ public function searchParentOf(ModelInterface $model)
         throw new DcGeneralInvalidArgumentException('Invalid condition, not in hierarchical mode!');
     }
 
+    /**
+     * Search the parent model for the given model being in a hierarchical structure.
+     *
+     * This method will determine the parent defined by the parent relationship and not the parent node of the
+     * hierarchical structure.
+     *
+     * @param ModelInterface $model The hierarchical model for search the parent model.
+     *
+     * @return ModelInterface|null
+     *
+     * @throws DcGeneralInvalidArgumentException It throws an exception if the configuration not passed.
+     * @throws DcGeneralInvalidArgumentException When the model does not originate from the child provider.
+     */
+    public function searchParentOfRootModel(ModelInterface $model): ?ModelInterface
+    {
+        $this->guardModelOriginatesFromProvider($model);
+        $this->guardRootProviderDefined();
+        $this->guardParentProviderDefined();
+
+        if ($this->rootProviderName !== $model->getProviderName()) {
```

Comment:
Is this check really required?

Reply:
As you can legally pass any model in here, IMO yes, it is required. Should be changed to utilize your new method `guardModelOriginatesFromProvider()` though.